atlasdb-api/src/test/java/com/palantir/atlasdb/keyvalue/api/AtlasLockDescriptorUtilsTest.java

```diff
@@ -0,0 +1,139 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.keyvalue.api;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.junit.Test;
+
+import com.google.common.collect.ImmutableList;
+import com.palantir.lock.AtlasCellLockDescriptor;
+import com.palantir.lock.AtlasRowLockDescriptor;
+import com.palantir.lock.LockDescriptor;
+import com.palantir.lock.StringLockDescriptor;
+
+import okio.ByteString;
+
+public class AtlasLockDescriptorUtilsTest {
+    private static final TableReference TABLE = TableReference.createFromFullyQualifiedName("test.table");
+    private static final byte[] NO_ZERO_ROW = new byte[] {1, 2, 3};
+    private static final byte[] NO_ZERO_COL = new byte[] {4, 5, 6};
+    private static final byte[] ROW_WITH_ZEROS = new byte[] {1, 0, 2, 0, 3};
+    private static final byte[] COL_WITH_ZEROS = new byte[] {4, 0, 0, 5, 6};
+    private static final byte[] START_WITH_ZERO = new byte[] {0, 1, 2, 3};
+    private static final byte[] END_WITH_ZERO = new byte[] {4, 5, 6, 0};
+
+    @Test
+    public void lockDescriptorWithNoZerosReturnsEmptyForCells() {
+        LockDescriptor descriptor = StringLockDescriptor.of("test");
+        assertThat(AtlasLockDescriptorUtils.candidateCells(descriptor)).isEmpty();
+    }
+
+    @Test
+    public void lockDescriptorWithOneZeroReturnsEmptyForCells() {
+        LockDescriptor descriptor = AtlasRowLockDescriptor.of(TABLE.getQualifiedName(), NO_ZERO_ROW);
+        assertThat(AtlasLockDescriptorUtils.candidateCells(descriptor)).isEmpty();
+    }
+
+    @Test
+    public void uniqueCellLockDescriptorIsDecodedCorrectlyForCells() {
+        LockDescriptor descriptor = AtlasCellLockDescriptor.of(TABLE.getQualifiedName(), NO_ZERO_ROW, NO_ZERO_COL);
+        assertThat(AtlasLockDescriptorUtils.candidateCells(descriptor))
+                .containsExactly(CellReference.of(TABLE, Cell.create(NO_ZERO_ROW, NO_ZERO_COL)));
+    }
+
+    @Test
+    public void rowWithNullsParsesAllCombinationsForCells() {
+        LockDescriptor descriptor = AtlasCellLockDescriptor.of(TABLE.getQualifiedName(), ROW_WITH_ZEROS, NO_ZERO_COL);
+
+        List<CellReference> expected = ImmutableList.of(
+                Cell.create(new byte[] {1}, new byte[] {2, 0, 3, 0, 4, 5, 6}),
+                Cell.create(new byte[] {1, 0, 2}, new byte[] {3, 0, 4, 5, 6}),
+                Cell.create(new byte[] {1, 0, 2, 0, 3}, new byte[] {4, 5, 6}))
+                .stream()
+                .map(cell -> CellReference.of(TABLE, cell))
+                .collect(Collectors.toList());
+
+        assertThat(AtlasLockDescriptorUtils.candidateCells(descriptor)).isEqualTo(expected);
```

Comment:
nit: `containsExactlyInAnyOrderElementsOf` as opposed to `isEqualTo` is probably what we care about? There could be something I'm missing.

Reply:
Since the return type is a list, I'd say the ordering is implicitly guaranteed (although not explicitly specified)